### DISCUSSION TITLE
Should pathname from useLocation be fully typed ?

### DISCUSSION THREAD

**Discussion by VdeVentura** [TanStack/router#1981] [Ideas]
Right now location is returning just a string, is there any reason not to have pathname being typed something similar to `ToPathOption` ?

**Comment by ViewableGravy** [comment]
I'd assume the reasoning behind this was that technically speaking, the returned value from useLocation is not actually the same as ToPathOptions. The pathname from useLocation is the builtLocation with search, params, hash, etc already added to the string, so it's not actually the path Id which is why it could be any string.

I agree it might be nice to provide the path ID from the useLocation hook as this has come up several times for me, but you can also access this by using `useMatches` and getting the last (most specific) match from the router if you don't actually know what route your component is on.

**Reply by VdeVentura** [reply]
100% right, I didn't really stop to think long enough before creating the discussion, I was only looking at my narrow use case where the pathname only had the path Id.

Thank you for taking the time tho.
